post-merge-review.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+set -e
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NC='\033[0m'
+REGEX_REF='^\(cherry picked from commit ([a-f0-9]{40})\)$'
+
+do_reference_check() {
+  if ! [[ $(git log --format=%B "$line^!" | grep -E "$REGEX_REF") =~ $REGEX_REF ]]; then
```

Comment:
You can avoid most `!` negations like this:

```suggestion
  [[ $(git log --format=%B "$line^!" | grep -E "$REGEX_REF") =~ $REGEX_REF ]] || {
     ...
  }
```

It also reads like plain English "do or die"

`!` sucks because it can't be copied to the interactive command line.

No big deal.

Reply:
`line` means `sha`?

Please avoid the global and pass it as `$1` argument, it's really not that much effort :-)

```shell
do_reference_check()
{
    local commit=$1
...
```

Reply:
`=~` is already a `grep` so it's not clear why there is also a `grep`.

Reply:
yea its a line from git which i have formatted to just return shas

Reply:
regarding grep iirc I was having trouble with global match, so grep gets me to the right line then does the capture groups

Reply:
I  think that's simply because `^` and `$` mean start of string and end of string. Not start of line and end of line. I did a bit of testing and including newlines in your regex should avoid the extra grep:

```shell
REGEX_REF='
\(cherry picked from commit ([a-f0-9]{40})\)
'
```